### PR TITLE
#162168429 Implement pagination on Travel Readiness Card

### PR DESCRIPTION
**Pull request by loicemeyo**
#### What does this PR do?
Implement pagination on Travel Readiness Card

#### Description of Task to be completed?

- Write functions to handle page change events
- Update css to match mockup
- Write tests for these functions

#### How should this be manually tested?

- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Change directory - cd travel_tool_front
- Checkout to current branch - git checkout ft-travel-readiness-pagination-162168429`
- Start the server - make start
- Install dependencies on the frontend - yarn install
- Start the server - yarn start
- Navigate to localhost:3000 in a browser
- Assign yourself the manager role and Travel Admin role so as to be able to approve your request in the database and view the dashboard page
- Navigate to the requests page and make a new request selecting your location as the destination.
- Navigate to the approvals page and approve your request
- Navigate to the dashboard page
- You should be able to view Travel Readiness card on the dashboard and pagination for the Card with icon buttons to `Previous` and `Next`
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[162168429](https://www.pivotaltracker.com/story/show/162168429)

#### Screenshots (if appropriate)
image
image

#### Questions:

### CODE REVIEW

File: src/components/AnalyticsReport/__tests__/AnalyticsReport.test.js [hunk 1]
```diff
@@ -47,10 +49,10 @@ describe('Test suite for Analytics Report Component', () => {
...
-    button.simulate('click');
-    expect(defaultProps.fetchReadiness).toHaveBeenCalled();
-  });
+  // it('should call fetchReadiness when export button is clicked', () => {
+  //   const mountWrapper = mount(<AnalyticsReport {...defaultProps} />);
+  //   const button = mountWrapper.find('.analyticsReport__export-button');
+  //   button.simulate('click');
+  //   expect(defaultProps.fetchReadiness).toHaveBeenCalled();
+  // });
 });
```
[the remark this answers is not in the document]
Comment (loicemeyo) [reply src/components/AnalyticsReport/__tests__/AnalyticsReport.test.js]: @DennisWanjiru Thanks for noting that. I have removed commented out code

File: src/components/AnalyticsReport/index.scss [hunk 2]
```diff
@@ -12,6 +50,10 @@ $white: #FFFFFF;
     color: $grey-color !important;
     line-height: 20px;
   }
+  .inactive {
+    height: 20px;	width: 60px;	color: #BCBCBC;	font-family: "DIN Pro";	font-size: 15px;	font-weight: 500;	line-height: 20px;	text-align: center;
```
Comment (matthewacha) [inline src/components/AnalyticsReport/index.scss]: format this line properly

File: src/components/TravelReadiness/__tests__/travelReadiness.test.js [hunk 3]
```diff
@@ -60,4 +68,35 @@ describe('Test suite for Travel Readiness Component', () => {
     exportBtn.simulate('click');
     expect(defaultProps.exportReadiness).toHaveBeenCalled();
   });
+  
+  it('should render the next page when next button is clicked', () => {
+    const newWrapper = mount(<TravelReadiness {...defaultProps} />);
+    const spy = sinon.spy(
```
Comment (matthewacha) [inline src/components/TravelReadiness/__tests__/travelReadiness.test.js]: remove this if you are not using it

File: src/components/TravelReadiness/__tests__/travelReadiness.test.js [hunk 4]
```diff
@@ -60,4 +68,35 @@ describe('Test suite for Travel Readiness Component', () => {
...
+      newWrapper.instance(),
+      'handleNextPage'
+    );
+    const nextbtn = newWrapper.find('#next');
+    nextbtn.simulate('click');
+    expect(defaultProps.fetchReadiness).toHaveBeenCalled();
+  });
+  it('should render the previous page when prevtious button is clicked', () => {
```
Comment (matthewacha) [inline src/components/TravelReadiness/__tests__/travelReadiness.test.js]: fix the typos

File: src/components/TravelReadiness/__tests__/travelReadiness.test.js [hunk 5]
```diff
@@ -60,4 +68,35 @@ describe('Test suite for Travel Readiness Component', () => {
...
+    const newProps = {...defaultProps };
+    newProps.readiness.pagination.currentPage = 2;
+    const newWrapper = mount(<TravelReadiness {...newProps} />);
+
+    const spy = sinon.spy(
```
Comment (matthewacha) [inline src/components/TravelReadiness/__tests__/travelReadiness.test.js]: remove this as well

File: src/components/TravelReadiness/index.jsx [hunk 6]
```diff
@@ -25,15 +61,26 @@ class TravelReadiness extends Component {
...
   getReadinessCSV = () => {
     const { exportReadiness } = this.props;
     exportReadiness({ type: 'file', });
 
   }
 
   render() {
-    const { readiness, renderNotFound, renderButton, renderSpinner } = this.props;
-    const { isLoading} = readiness;
+    const { readiness, renderNotFound} = this.props;
```
Comment (matthewacha) [inline src/components/TravelReadiness/index.jsx]: leave a space before the closing curly bracket

File: src/components/TravelReadiness/index.jsx [hunk 7]
```diff
@@ -25,15 +61,26 @@ class TravelReadiness extends Component {
       </div>
     );
   }
+
+  renderPagination = (readiness, dataCount) => {
+    return (
+      <div className="pag">
+        <button id="prev" type="button" style={{ border:0, backgroundColor:'transparent', borderColor:'transparent' }} onClick={this.handlePrevPage} className={readiness.pagination && readiness.pagination.currentPage !== 1 ? 'active': 'inactive'}>Previous</button>
+        <p className="showing-results">{`Showing ${readiness.readiness.length} of ${dataCount.value} results`}</p>
+        <button id="next" type="button" onClick={this.handleNextPage} className={readiness.pagination && !this.hasNextPage() ? 'active': 'inactive'}>Next</button>
```
Comment (matthewacha) [inline src/components/TravelReadiness/index.jsx]: these lines are too long

Comment (loicemeyo) [reply src/components/TravelReadiness/index.jsx]: Thanks @matthewacha for pointing out these. I have implemented all the changes requested

File: src/components/TravelReadiness/index.jsx (hunk 6 above)
Comment (loicemeyo) [reply src/components/TravelReadiness/index.jsx]: Thanks @matthewacha for noting this. I have fixed it

File: src/components/TravelReadiness/__tests__/travelReadiness.test.js (hunk 5 above)
Comment (loicemeyo) [reply src/components/TravelReadiness/__tests__/travelReadiness.test.js]: Thanks @matthewacha for noting this. I have removed unused code

**Comment by amarachukwu-agbo** [comment]
@loicemeyo, please fix the typo error in your commit. Also, please resolve merge conflicts on this.

**Comment by loicemeyo** [comment]
> @loicemeyo, please fix the typo error in your commit. Also, please resolve merge conflicts on this.

@amarachukwu-agbo  Thank you, I have resolved conflicts and fixed the typo

**Comment by amarachukwu-agbo** [comment]
@loicemeyo , the PR still has merge conflicts. Also endeavour to fix the codeclimate issue

**Comment by amarachukwu-agbo** [comment]
While testing the PR, I realised that the part that displays the number of results has an issue. The first page shows '9 of 11 results' when I click on the next button, it shows '2 of 11 results'. I think this is kind of confusing and we should use pages instead ie 'Showing page 1 of 2' and 'Showing page 1 of 2'
This is @asamisellah's current implementation and I think you both can sync on this for consistency.

screenshot 2018-11-28 at 2 57 10 pm
screenshot 2018-11-28 at 2 57 16 pm

File: src/components/TravelReadiness/__tests__/travelReadiness.test.js [hunk 8]
```diff
@@ -15,16 +16,23 @@ const defaultProps = {
...
       arrivalDate: '2018-10-05T00:00:00.000Z'
     }],
+    pagination:  {
+      pageCount: 2,
+      currentPage: 1,
+      dataCount: 13
+    },
   },
   fetchReadiness: jest.fn(),
+  // handleNextPage: jest.fn(),
```
Comment (amarachukwu-agbo) [inline src/components/TravelReadiness/__tests__/travelReadiness.test.js]: Please remove this unused line

File: src/components/TravelReadiness/__tests__/travelReadiness.test.js [hunk 9]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { shallow, mount } from 'enzyme';
+import sinon from 'sinon';
```
Comment (amarachukwu-agbo) [inline src/components/TravelReadiness/__tests__/travelReadiness.test.js]: I don't see sinon be used in this test. If that is the case we can have it removed

**Comment by amarachukwu-agbo** [comment]
I also noticed some styling issues with the left side bar. The dashboard text becomes smaller when the dashboard link is active. Also the sub-links appear truncated for items that have sub-links like 'Requests', 'Residence' and 'Settings'

**Comment by loicemeyo** [comment]
Hi @amarachukwu-agbo The current implementation is according to the mockup provided for the card. I don't know why the behaviour on the sidebar is like so. It appears intermittently on my browser but I've included a fix for this. Please let me know if it works well on your side now

**Comment by amarachukwu-agbo** [comment]
@loicemeyo , please resolve merge conflicts

**Comment by amarachukwu-agbo** [comment]
LGTM

**Pull request merged**